src/System.Security.Cryptography.Algorithms/src/Internal/Cryptography/Rfc4050KeyFormatter.cs

```diff
@@ -0,0 +1,399 @@
+            }
+            if (!navigator.MoveToFirstAttribute() || navigator.Name != ValueAttribute || String.IsNullOrEmpty(navigator.Value))
+            {
+                throw new ArgumentException(SR.Cryptography_MissingPublicKey);
+            }
+
+            BigInteger x = BigInteger.Parse(navigator.Value, CultureInfo.InvariantCulture);
+            navigator.MoveToParent();
+
+            // Then the y parameter
+            if (!navigator.MoveToNext(XPathNodeType.Element) || navigator.Name != YElement)
+            {
+                throw new ArgumentException(SR.Cryptography_MissingPublicKey);
+            }
+            if (!navigator.MoveToFirstAttribute() || navigator.Name != ValueAttribute || String.IsNullOrEmpty(navigator.Value))
+            {
+                throw new ArgumentException(SR.Cryptography_MissingPublicKey);
+            }
+
+            BigInteger y = BigInteger.Parse(navigator.Value, CultureInfo.InvariantCulture);
+
+            byte[] xBytes = x.ToByteArray();
+            byte[] yBytes = y.ToByteArray();
+
+            int xLen = xBytes.Length;
+            int yLen = yBytes.Length;
+
+            // If the last byte of X is 0x00 that's a padding byte by BigInteger to indicate X is
+            // a positive number with the highest bit in the most significant byte set. We can't count
+            // that in the length of the number.
+            if (xLen > 0 && xBytes[xLen - 1] == 0)
+            {
+                xLen--;
+            }
+
+            // Ditto for Y.
+            if (yLen > 0 && yBytes[yLen - 1] == 0)
+            {
+                yLen--;
+            }
+
+            // Q.X and Q.Y have to be the same length.  They ultimately have to be the right length for the curve,
+            // but that requires more knowledge than we have. So we'll ask the system. If it doesn't know, just make
+            // them match each other.
+            int requiredLength = Math.Max(xLen, yLen);
+
+            try
+            {
+                using (ECDsa ecdsa = ECDsa.Create(parameters.Curve))
+                {
+                    // Convert the bit value of keysize to a byte value.
+                    // EC curves can have non-mod-8 keysizes (e.g. 521), so the +7 is really necessary.
+                    int curveLength = (ecdsa.KeySize + 7) / 8;
+
+                    // We could just use this answer, but if the user has formatted the input to be
+                    // too long, maybe they know something we don't.
+                    requiredLength = Math.Max(requiredLength, curveLength);
+                }
+            }
+            catch (ArgumentException) { /* Curve had invalid data, like an empty OID */ }
+            catch (CryptographicException) { /* The system failed to generate a key for the curve */ }
+            catch (NotSupportedException) { /* An unknown curve type was requested */ }
+
+            // There is a chance that the curve is known to Windows but not allowed for ECDH
```

Comment:
Apparently I made a logic-o in the netfx version if this.  "not allowed for ECDH" should be "only allow for ECDH"